extensions/notebook/src/dialog/managePackages/addNewPackageTab.ts

```diff
@@ -0,0 +1,264 @@
+				this.dialog.showErrorMessage(this.NoVersionsFoundError);
+				return;
+			}
+
+			await this.newPackagesName.updateProperties({
+				value: packageName
+			});
+			await this.newPackagesVersions.updateProperties({
+				values: pipPackage.versions,
+				value: pipPackage.versions[0],
+			});
+			await this.newPackagesSummary.updateProperties({
+				value: pipPackage.summary
+			});
+
+			// Only re-enable install on success
+			await this.packageInstallButton.updateProperties({ enabled: true });
+		} catch (err) {
+			this.dialog.showErrorMessage(utils.getErrorMessage(err));
+
+			await this.newPackagesName.updateProperties({
+				value: this.InvalidTextPlaceholder
+			});
+			await this.newPackagesVersions.updateProperties({
+				value: this.InvalidTextPlaceholder,
+				values: [this.InvalidTextPlaceholder],
+			});
+			await this.newPackagesSummary.updateProperties({
+				value: this.InvalidTextPlaceholder
+			});
+		} finally {
+			await this.toggleNewPackagesFields(true);
+		}
+	}
+
+	private async fetchPypiPackage(packageName: string): Promise<PipPackageData> {
+		return new Promise<PipPackageData>((resolve, reject) => {
+			request.get(`https://pypi.org/pypi/${packageName}/json`, { timeout: 10000 }, (error, response, body) => {
+				if (error) {
+					return reject(error);
+				}
+
+				if (response.statusCode === 404) {
+					return reject(this.PackageNotFoundError);
+				}
+
+				if (response.statusCode !== 200) {
+					return reject(response.statusMessage);
+				}
+
+				let packagesJson = JSON.parse(body);
+				let versionKeys = Object.keys(packagesJson.releases);
+				versionKeys = versionKeys.filter(versionKey => {
+					let releaseInfo = packagesJson.releases[versionKey];
+					return Array.isArray(releaseInfo) && releaseInfo.length > 0;
+				});
+				versionKeys.sort((first, second) => {
+					// sort in descending order
+					let firstVersion = first.split('.').map(numStr => Number.parseInt(numStr));
+					let secondVersion = second.split('.').map(numStr => Number.parseInt(numStr));
+
+					// If versions have different lengths, then append zeroes to the shorter one
+					if (firstVersion.length > secondVersion.length) {
+						let diff = firstVersion.length - secondVersion.length;
+						secondVersion = secondVersion.concat(new Array(diff).fill(0));
+					} else if (secondVersion.length > firstVersion.length) {
+						let diff = secondVersion.length - firstVersion.length;
+						firstVersion = firstVersion.concat(new Array(diff).fill(0));
+					}
+
+					for (let i = 0; i < firstVersion.length; ++i) {
+						if (firstVersion[i] > secondVersion[i]) {
+							return -1;
+						} else if (firstVersion[i] < secondVersion[i]) {
+							return 1;
+						}
+					}
+					return 0;
+				});
+
+				resolve({
+					name: packageName,
+					versions: versionKeys,
+					summary: packagesJson.info.summary
+				});
+			});
+		});
+	}
+
+	private async doPackageInstall(): Promise<void> {
```

Comment:
Could we wrap this in the azdata tasks API? At some point this should have support for progress messages etc., and that's the standard way to do a potentially long-running task like this.